tests/clone-flags.c

```diff
@@ -23,11 +23,26 @@
 static const int child_exit_status = 42;
 static pid_t pid;
 
+static void
+retrieve_userns(pid_t pid, char *userns_buf, size_t userns_buflen)
+{
+	char *fname = xasprintf("/proc/%d/ns/user", pid);
+	int rc = readlink(fname, userns_buf, userns_buflen - 1);
+	if ((unsigned int) rc >= userns_buflen)
+		perror_msg_and_fail("readlink");
+	userns_buf[rc] = '\0';
+}
+
 static pid_t
-wait_cloned(pid_t pid, int sig, const char *text)
+wait_cloned(pid_t pid, int sig, const char *text, char *userns_buf, size_t userns_buflen)
 {
-	if (pid < 0)
+	if (pid < 0) {
+		if (userns_buf)
+			perror_msg_and_skip("clone %s", text);
```

Comment:
Creating user namespaces can be disabled, and it would be undesirable to skip the whole test because of that.
Let's just return pid here.

Reply:
fixed locally.